### PR TITLE
update /profile/choosetopics

### PR DESCRIPTION
**Pull request by scuffed-developer**
Fix from previous pr #20

- initializing small to do to make pull request
- update : refactor and added loading state
- update : topics now directly fetched from rtdb
- cleaning console log (#20)
- fix : order now works

### CODE REVIEW

**Comment by arunstop** [comment]
Closing issue:#21. Great job! What went wrong btw? @scuffed-developer

File: utils/services/network/RtdbModules.ts [hunk 1]
```diff
@@ -90,12 +94,17 @@ export async function rtdbTopicGet(): Promise<string[]> {
...
-    if(snapshot.exists()){
-      // console.log(snapshot.val());
-      return snapshot.val();
+  const qq = query(rr, orderByChild("name"));
+  const res = await get(qq).then((snapshot) => {
+    if (snapshot.exists()) {
+      
+      const data: string[] = [];
+      // using `forEach` because normal `val` won't guarantee the ordering request
+      snapshot.forEach((e) => {
```
Comment (scuffed-developer) [inline utils/services/network/RtdbModules.ts:103]: @arunstop So basically we cannot use just play `.val()` from the snapshot. Cuz it said in the documentation (when i hover on my IDE) that `.val()` doesn't guarantee the requested order. The documentation also said exactly why you should use `.forEach()` instead, cuz it guarantees the requested order.


Pretty weird technicality i would say, but i guess it must have been designed exactly like that for performance reason or something. After all they are Google engineers who alchemized all of this! I was so stunted, weirded and all other negative feeling when i was struggling with this.

The bright side is #TodayILearned :tada:

Comment (arunstop) [reply utils/services/network/RtdbModules.ts:103]: Alright! Today i definitely learned as well then! Thanks for the info! :+1:

**Review by arunstop — APPROVED** [review]

**Comment by arunstop** [comment]
reverting the #20 please approve #24

**Pull request closed without merging**